catalog/dags/providers/provider_api_scripts/auckland_museum.py

```diff
@@ -0,0 +1,201 @@
+                                    "to": self.date_to,
+                                }
+                            }
+                        },
+                    ]
+                }
+            }
+        }
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        # Return default query params on the first request
+        # primaryRepresentation contain a image url for each data
+        # "+" is a query string syntax for must be present
+        # copyright:CC state Creative Commons Attribution 4.0
+        return {
+            "size": "2000",
+            "from": self.batch_start,
+        }
+
+    def get_batch_data(self, response_json):
+        # Takes the raw API response from calling `get` on the endpoint, and returns
+        # the list of records to process.
+        if response_json:
+            return response_json.get("hits", {}).get("hits")
+        return None
+
+    def get_should_continue(self, response_json):
+        # Do not continue if we have exceeded the total amount of data
+        self.batch_start += self.batch_limit
+        if self.batch_start >= self.total_amount_of_data:
+            logger.info(
+                "The final amount of data has been processed. Halting ingestion."
+            )
+            return False
+
+        return True
+
+    def get_media_type(self, record: dict):
+        return IMAGE
+
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        # check if _id is empty then foreign_landing_url and
+        # foreign_identifier doesn't exist
+
+        if not data.get("_id"):
+            return None
+
+        url_parameter = data.get("_id").split("id/")[-1].replace("/", "-")
+        foreign_landing_url = f"{LANDING_URL}{url_parameter}"
+
+        foreign_identifier = data.get("_id").split("/")[-1]
+
+        information = data.get("_source", {})
+
+        if not (url := information.get("primaryRepresentation")):
+            return None
+
+        thumbnail_url = f"{url}?rendering=thumbnail.jpg"
+        license_info = self.DEFAULT_LICENSE_INFO
+
+        creator = (
+            information.get("dc_contributor")[0]
+            if information.get("dc_contributor", [])
+            else None
+        )
+
+        appellation = information.get("appellation", {})
+        title = (
+            appellation.get("Primary Title")[0]
+            if appellation.get("Primary Title")
+            else None
+        )
+        meta_data = self._get_meta_data(information)
+
+        return {
+            "foreign_landing_url": foreign_landing_url,
+            "foreign_identifier": foreign_identifier,
+            "url": url,
+            "license_info": license_info,
+            "thumbnail_url": thumbnail_url,
+            "creator": creator,
+            "title": title,
+            "meta_data": meta_data,
+        }
+
+    def _get_meta_data(self, object_json: dict) -> dict | None:
```

Comment:
```suggestion
    @staticmethod
    def _get_meta_data(self, object_json: dict) -> dict | None:
```
This method does not use any data from the class instance (`self`), so it's better to make it static (as Pycharm IDE suggests :) )